src/index.js

```diff
@@ -262,30 +262,18 @@ class SignatureFactory {
     this.method = method;
   }
 
-  getCanonicalHeaders () {
-    return {
-      date: `date:${this.dates.usbldate}`,
-      host: `host:${this.host}\n`
-    };
-  }
-
   hash (string, encoding) {
     return crypto.createHash('sha256').update(string, 'utf8').digest(encoding);
   }
 
   canonicalString () {
-    const queryStr = this.queryParameters;
-    const bodyHash = this.hash('', 'hex');
-    const canonicalHeaders = this.getCanonicalHeaders();
-
     return [
-      this.method || 'GET',
-      this.url,
-      queryStr,
-      canonicalHeaders.date,
-      canonicalHeaders.host,
-      'date;host',
-      bodyHash
+      this.method || 'GET',               // HTTPRequestMethod
+      this.url,                           // CanonicalURI
+      this.queryParameters,               // CanonicalQueryString
+      `host:${this.host}`+ '\n' + `x-usbl-date:${this.dates.longdate}`+ '\n', // CanonicalHeaders
```

Comment:
Hm, this would also not work. From your perspective you want;
`return [
  ...,
  this.queryParameters,
  `host:${this.host}`,
  `x-usbl-date:${this.dates.longdate}\n`,
  'host;x-usbl-date',
  ...
];
`
This is weird, as the comma joins and adds a `\n` between the properties of the CanonicalString. Now you use that functionality to add a `\n` for a header inside a property of the CanonicalString. That break in consistency makes it more unreadable.

The explanation can be moved to the top, no problem. I copied this practise from usabilla dev PHP coding.